### PR TITLE
#184842353 Implement Newsletter Signup

### PR DESCRIPTION
**Pull request by khevin2**
#### What does this PR do?
This PR adds functionalities so that a user can
- Sign up to newsletters
- Receive email about the status of subscription by email
- Verify their email so that a user can receive periodic newsletters

#### Description of Task to be completed?
- Allow user to subscribe, unsubscribe and verify his email address to receive newsletters
- Allow admin to suspend, delete subscribers, get subscribers or by email
#### How should this be manually tested?
- Clone this repo git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with .env-example
- Install all the dependencies by running npm install
- Then run the project by typing npm start
#### To test this PR,
- In your **e-comm-team-amigos-bn** directory, open terminal and run `git fetch origin pull/55/head:ft-newsletter-signup-184842353`
- Checkout to **ft-newsletter-signup-184842353** branch
- Run `npm install`
- Run `npm run dev`
- In your working directory terminal, run migrations and seeders. 
- Test endpoints

#### Endpoints to be tested:
- `POST /newsletter`
- `GET /newsletter/verify`
- `GET /newsletter/unsubscribe`
- `DELETE /newsletter/admin`
- `POST /newsletter/admin`
- `GET /newsletter/admin`
- `GET /newsletter/admin/{email}`

#### Any background context you want to provide?
- Have Node Js installed in your local machine
- Have Postgres
- API tester like Postman

#### What are the relevant pivotal tracker?

Story ID: [#184842353](https://www.pivotaltracker.com/story/show/184842353)

#### Screenshots 
Screenshot from 2023-04-03 13-55-33

Screenshot from 2023-04-03 13-55-00

### CODE REVIEW

File: src/utils/newsletter.mail.js [hunk 1]
```diff
@@ -0,0 +1,769 @@
+import nodemailer from "nodemailer";
+import fs from "fs";
+import path from "path";
```
Comment (Emile-SHUMBUSHO) [inline src/utils/newsletter.mail.js]: these values seems to be not used in your codes, just remove it

**Review by AbdulKhaliq59 — APPROVED** [review]
LGTM

**Comment by Emile-SHUMBUSHO** [comment]
file src/utils/newsletter.mail.js should not be located in utils directory according to it's serving. it's a good practice to move it into src/helpers directory.

**Comment by Emile-SHUMBUSHO** [comment]
This PR missed some functionality from develop remember to rebase otherwise this PR won't be merged.

**Review by GasanaJr — CHANGES_REQUESTED** [review]
You have circleci fails. fix the tests

**Review by Rwibutso — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]
LGTM

**Review by ericndungutse — APPROVED** [review]

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Comment by Emile-SHUMBUSHO** [comment]
Resolve conflicts otherwise LGTM

**Review by Mystica52 — APPROVED** [review]

**Review by Mystica52 — COMMENTED** [review]
circle ci is failing

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Pull request merged**